grammars/tree-sitter-javascript.cson

```diff
@@ -132,7 +132,13 @@ scopes:
   'null': 'constant.language.null'
   'true': 'constant.language.boolean.true'
   'false': 'constant.language.boolean.false'
-  'comment': 'comment.block'
+  'comment': [
+    {
+      match: "^//",
```

Comment:
Line comments can appear anywhere on a line; this only handles cases where it's at the beginning of a line.
We need tests that explicitly handles various line comment scenarios.

Reply:
@thomasjo The regex here refers to the start of the comment node, which should always be `//` if it's a simple line comment. It will catch comments at the end of a line

I'll add more tests though

Reply:
Aaaha, thanks for clarifying that :bowtie:

Reply:
Actually, I didn't add tests the first time because none exist yet. I guess I'll (try and) make a new set for Tree-sitter.

Reply:
Emphasis on try. There's no point in testing the tree; that's done by the parser's package. And I'm having trouble getting the scopes.

Reply:
I'm ok with not having tests on these scope mappings. The actual *parsing* is tested very exhaustively, but testing these kinda feels like testing CSS.

If you have a strong desire to add tests, I'm ok with it, but I'm also cool merging this as-is.

Reply:
- I think tests are a good idea for the Atom specific behaviour, but was [unable to get it working](https://github.com/atom/atom/issues/18443)
- This issue can be raised on almost all the repos though, as they all treat comments as blocks